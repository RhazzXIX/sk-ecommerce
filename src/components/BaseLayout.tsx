import Link from "next/link";

export default function LayoutBase({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <>
      <nav>
        <Link href={"/"}>Home</Link>
        <Link href={"/list"}>V-list</Link>
      </nav>
      {children}
      <footer>
        <p>
          <Link href={"/about"}>About</Link>
          &gt;&gt;&gt;
        </p>
      </footer>
    </>
  );
}
