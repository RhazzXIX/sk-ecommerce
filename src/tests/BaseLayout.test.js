import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import BaseLayout from "../components/BaseLayout";

describe("BaseLayout component", () => {
  it("Renders on screen", () => {
    const { container } = render(<BaseLayout />);

    expect(container).toMatchSnapshot();
  });
  it("Renders it's children", () => {
    render(
      <BaseLayout>
        <main>
          <h1>Hi! I&apos;m the child!</h1>
        </main>
      </BaseLayout>
    );
    expect(
      screen.getByRole("heading", { name: "Hi! I'm the child!" })
    ).toBeInTheDocument();
  });
});
