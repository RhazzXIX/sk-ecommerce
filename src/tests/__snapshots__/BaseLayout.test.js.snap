// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout component Renders on screen 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/list"
    >
      V-list
    </a>
  </nav>
  <footer>
    <p>
      <a
        href="/about"
      >
        About
      </a>
      &gt;&gt;&gt;
    </p>
  </footer>
</div>
`;
